src/main.ts

```diff
@@ -0,0 +1,51 @@
+import * as core from "@actions/core";
+import * as github from "@actions/github";
+import { Octokit } from "@octokit/rest";
+import { Value } from "@sinclair/typebox/build/cjs/value";
+import { envSchema, pluginSettingsSchema, PluginInputs } from "./types";
+import { plugin } from "./plugin";
+
+/**
+ * How a GitHub action executes the plugin.
+ */
+export async function run() {
+  const payload = github.context.payload.inputs;
+
+  const env = Value.Decode(envSchema, process.env);
+  const settings = Value.Decode(pluginSettingsSchema, JSON.parse(payload.settings));
```

Comment:
Should have a validation step as well

Reply:
Do you mean like this, as this happens inside `plugin.ts` already and the kernel validates the config for bot and all plugins if I'm not mistaken?

```ts
  if (context.eventName === "issue_comment.created") {
    await doSomething(context);
  } else {
    context.logger.error(`Unsupported event: ${context.eventName}`);
  }
```

- https://github.com/ubiquibot/assistive-pricing/blob/development/src/index.ts
- https://github.com/ubiquibot/assistive-pricing/blob/development/src/run.ts

I was looking at this as a reference for what you meant and there is no validation beyond confirming the payload event type

Reply:
I believe we should add checks there later on, it's always good practice and we can display detailed errors.
There is validation in [this plugin](https://github.com/ubiquibot/conversation-rewards/blob/2f3c2ee229400031e1fd95324d91677eda84925e/src/configuration/incentives.ts#L36).